src/fractal_faim_hcs/MetaSeriesUtils_dask.py

```diff
@@ -0,0 +1,302 @@
+
+import dask.array as da
+import numpy as np
+import pandas as pd
+from faim_hcs.MetaSeriesUtils import (
+    _build_ch_metadata,
+    compute_z_sampling,
+    get_img_YX,
+    montage_grid_image_YX,
+)
+from faim_hcs.UIntHistogram import UIntHistogram
+from numpy._typing import ArrayLike
+from tifffile import tifffile
+
+
+
+def get_well_image_CZYX_lazy(  # noqa: C901
+    well_files: pd.DataFrame,
+    channels: list[str],
+    assemble_fn: Callable = montage_grid_image_YX,
+) -> tuple[ArrayLike, list[UIntHistogram], list[dict], dict]:
+    """Assemble image data for the given well-files."""
+    # TODO: This function assumes that there are no gaps in data (i.e. that
+    # there are no images missing to fill a full FCZ-stack)
+    # -> maybe implement checks
+
+    planes = sorted(well_files["z"].unique(), key=int)
+
+    fields = sorted(
+        well_files["field"].unique(),
+        key=lambda s: int(re.findall(r"(\d+)", s)[0])
+    )
+
+    # Create an empty np array to store the filenames in the correct structure
+    fn_dtype = f"<U{max([len(fn) for fn in well_files['path']])}"
+    fns_np = np.zeros(
+        (len(fields), len(channels), len(planes)),
+        dtype=fn_dtype,
+    )
+
+    # Store fns in correct position
+    for s, field in enumerate(fields):
+        field_files = well_files[well_files["field"] == field]
+        for c, channel in enumerate(channels):
+            channel_files = field_files[field_files["channel"] == channel]
+            for z, plane in enumerate(planes):
+                plane_files = channel_files[channel_files["z"] == plane]
+                if len(plane_files) == 1:
+                    fns_np[s, c, z] = list(plane_files["path"])[0]
+                elif len(plane_files) > 1:
+                    raise RuntimeError("Multiple files found for one FCZ")
+
+    # distinguish between channels with
+    #   - all z-planes
+    #   - only one z-plane (stored in first plane)
+    #   - no z-planes (only projections)
+    channels_full_planes = []
+    channels_one_plane = []
+    channels_empty = []
+    for c in range(fns_np.shape[1]):
+        if '' not in fns_np[:,c,:]:
+            channels_full_planes.append(c)
+        elif '' not in fns_np[:,c,0]:
+            channels_one_plane.append(c)
+        else:
+            channels_empty.append(c)
+
+
+    # LOAD STAGE POSITIONS:
+    # TODO: maybe only read z-positions
+    def _da_read_stage_positions(x):
+        fn = x.flatten()[0]
+        if fn=='':
+            raise RuntimeError('File not found. This channel probably either only has projections or only one plane.')
+        ms_metadata = load_metaseries_tiff_metadata(fn)
+        x_pos = ms_metadata["stage-position-x"]
+        y_pos = ms_metadata["stage-position-y"]
+        z_pos = ms_metadata["z-position"]
+        output = np.array([x_pos, y_pos, z_pos], dtype="float64")
+        newshape = x.shape + output.shape
+        return np.reshape(output, newshape)
+
+    fns_da = da.from_array(
```

Comment:
What is fns_da actually? We're creating it here, but then overwriting it again in line 148